src/par_iter/find_first_last.rs

```diff
@@ -0,0 +1,183 @@
+use std::cell::Cell;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use super::internal::*;
+use super::*;
+use super::len::*;
+
+// The consumer for find_first/find_last has fake indexes representing the lower
+// and upper bounds of the "range" of data it consumes. This range does not
+// correspond to indexes from the consumed iterator but rather indicate the
+// consumer's position relative to other consumers. The purpose is to allow a
+// consumer to know it should stop consuming items when another consumer finds a
+// better match.
+
```

Comment:
Total nit: I'd like a `//` here to indicate that these are two paragraphs in the same comment.